crates/sui-types/src/crypto.rs

```diff
@@ -1550,16 +1551,50 @@ impl<'a> VerificationObligation<'a> {
 
     pub fn verify_all(self) -> SuiResult<()> {
         let mut pks = Vec::with_capacity(self.public_keys.len());
-        for pk in self.public_keys {
+        for pk in self.public_keys.clone() {
             pks.push(pk.into_iter());
         }
         AggregateAuthoritySignature::batch_verify(
             &self.signatures.iter().collect::<Vec<_>>()[..],
             pks,
             &self.messages.iter().map(|x| &x[..]).collect::<Vec<_>>()[..],
         )
-        .map_err(|e| SuiError::InvalidSignature {
-            error: format!("Failed to batch verify aggregated auth sig {}", e),
+        .map_err(|e| {
+            let message = format!(
+                "pks: {:?}, messages: {:?}, sigs: {:?}",
+                &self.public_keys,
+                self.messages
+                    .iter()
+                    .map(Base64::encode)
+                    .collect::<Vec<String>>(),
+                &self
+                    .signatures
+                    .iter()
+                    .map(|s| Base64::encode(s.as_ref()))
+                    .collect::<Vec<String>>()
+            );
+
+            let chunk_size = 2048;
+
+            // This error message may be very long, so we print out the error in chunks of to avoid
+            // hitting a max log line length on the system.
+            for (i, chunk) in message
+                .as_bytes()
+                .chunks(chunk_size)
+                .map(std::str::from_utf8)
+                .enumerate()
+            {
+                warn!(
+                    "Failed to batch verify aggregated auth sig: {} (chunk {}): {}",
```

Comment:
need to have tx_digest in every print. Otherwise it's difficult for node operators to collect them.

Reply:
We've asked people to grep tx_digest

Reply:
That is handled by the debug span, re https://github.com/MystenLabs/sui/pull/10895#issuecomment-1508310473.

Reply:
Actually, that might only be true for debug! and not warn!.